src/middlewared/middlewared/plugins/container/lifecycle.py

```diff
@@ -44,19 +47,21 @@ def stop(self, id_, options):
             self.middleware.libvirt_domains_manager.containers.destroy(pylibvirt_container)
 
     @private
-    def pylibvirt_container(self, container):
+    def pylibvirt_container(self, container, check_ds=False):
         container = container.copy()
         container.pop("id", None)
         container.pop("status", None)
 
         dataset = container.pop("dataset")
-        datasets = self.middleware.call_sync(
-            'zfs.resource.query_impl', {'paths': [dataset], 'properties': ['mountpoint']}
-        )
-        if not datasets:
-            raise CallError(f"Dataset {dataset!r} not found", errno.ENOTDIR)
+        pool = dataset.split("/")[0]
+        container["root"] = f"/mnt/{container_instance_dataset_mountpoint(pool, container['name'])}"
+        if check_ds:
+            datasets = self.middleware.call_sync(
```

Comment:
Why are you querying the `mountpoint` property if you're checking for its existence?